server/model/monitor.js

```diff
@@ -1018,18 +1016,35 @@ class Monitor extends BeanModel {
             }
 
             return res;
-        } catch (e) {
+        } catch (error) {
+
+            /**
+             * Make a single attempt to obtain an new access token in the event that
+             * the recent api request failed for authentication purposes
+             */
+            if (this.auth_method === "oauth2-cc" && error.response.status === 401 && !options.isReattempt) {
+                this.oauthAccessToken = await this.makeOidcTokenClientCredentialsRequest();
+                let oauth2AuthHeader = {
+                    "Authorization": this.oauthAccessToken.token_type + " " + this.oauthAccessToken.access_token,
+                };
+                options.headers = { ...(options.headers),
+                    ...(oauth2AuthHeader)
+                };
+                options.isReattempt = true;
+                return this.makeAxiosRequest(options, true);
```

Comment:
It works but I would't use `isReattempt` here because uptime-kuma already has a built in logic for this. See line 1039 with `!finalCall` in the if statement. Yours works the the same but is an additional method of archiving the same thing,

Reply:
Absolutely, resolved in the next commit